spire/public/api.py

```diff
@@ -76,62 +72,75 @@ async def version() -> VersionResponse:
     return VersionResponse(version=SPIRE_PUBLIC_VERSION)
 
 
-@app_public.get("/check", tags=["public journals"])
+@app_public.get("/", tags=["public journals"])
+async def list_public_journals_handler(
+    user_id: UUID = Query(...),
+    db_session: Session = Depends(yield_db_read_only_session),
+) -> BugoutJournals:
+    """
+    List journals with public access.
+    """
+    try:
+        public_user = actions.get_public_user(db_session=db_session, user_id=user_id)
+        result: BugoutJournals = bugout_api.list_journals(
+            token=public_user.restricted_token_id
```

Comment:
Should we restrict this to list only journals whose journal id is in `public_journals.journal_id`?

Reply:
I do not think so, if this user (we operate as autogenerated public user) has access to journal, then someone granted permissions to read a journal.